zebra-chain/src/parallel/tree.rs

```diff
@@ -79,8 +79,19 @@ impl NoteCommitmentTrees {
         let mut sapling_result = None;
         let mut orchard_result = None;
 
+        // Note: Only updating the note commitment trees when there are more notes that
+        //       need to be appended to the tree prevents unnecessarily duplicating note commitment
+        //       tree data in the non-finalized state.
```

Comment:
There's always another Arc that owns these, the value will always be cloned. That's fine, we want to always clone the inner value, otherwise it would be updating the note commitment tree for the previous chain tip.

```suggestion
```